staging/src/k8s.io/metrics/pkg/apis/external_metrics/v1beta1/types.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// a list of values for a given metric for some set of objects
+type ExternalMetricValueList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// the value of the metric across the described objects
+	Items []ExternalMetricValue `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// a metric value for some object
+type ExternalMetricValue struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// the name of the metric
+	MetricName string `json:"metricName" protobuf:"bytes,2,name=metricName"`
+
+	// labels set identifying the value within metric
```

Comment:
This comment is not very helpful - I have no idea what to expect to see here.

In general we avoid maps, but I have no context to evaluate this field to tell whether it is OK or not.